cupy/testing/helper.py

```diff
@@ -1021,14 +1037,21 @@ def for_orders(orders, name='order'):
     def decorator(impl):
         @functools.wraps(impl)
         def test_func(self, *args, **kw):
+            exists_true_case = False
             for order in orders:
                 try:
                     kw[name] = order
-                    impl(self, *args, **kw)
+                    ret = impl(self, *args, **kw)
+                    if ret in (True, None):
+                        exists_true_case = True
                 except Exception:
                     print(name, 'is', order)
                     raise
 
+            if not exists_true_case:
+                raise AssertionError(
+                    'An accepted error was raised in all orders. '
```

Comment:
Just a nitpick
Maybe "all combinations" instead of "all orders"?

Feel free to ignore this.